### PR TITLE
ci: fix github actions

### PR DESCRIPTION
**Pull request by grondo**
I noticed the Github workflow for this repo was running the pr-validator as part of the matrix job, which is unnecessary, this action only should be run once.

While I was working on it, I also moved the `coverage` check into the github workflows so that Travis-CI will no longer be required for this repo.

However, after doing that it looks like more work is needed. The unittest doesn't actually run any tests, so the current GH actions aren't checking anything. It also looks like pylint is catching errors and  `black` is reformatting code, but that isn't caught as an error in the actions(?)

https://github.com/flux-framework/flux-accounting/runs/1502453306?check_suite_focus=true#step:8:61

```
./config/tap-driver.py:8:1: E302 expected 2 blank lines, found 1
./config/tap-driver.py:9:29: E261 at least two spaces before inline comment
./config/tap-driver.py:25:1: E305 expected 2 blank lines after class or function definition, found 1
./t/pycotap/pycotap/__init__.py:19:1: E302 expected 2 blank lines, found 1
./t/pycotap/test/test.py:5:11: E401 multiple imports on one line
./t/pycotap/test/test.py:9:1: E402 module level import not at top of file
./t/pycotap/test/test.py:10:1: E402 module level import not at top of file
./t/pycotap/test/test.py:17:1: E402 module level import not at top of file
./t/pycotap/example/test_example.py:4:11: E401 multiple imports on one line
./t/pycotap/example/test_example.py:8:1: E402 module level import not at top of file
./t/pycotap/example/test_example.py:9:1: E402 module level import not at top of file
./src/bindings/python/flux/accounting/flux-account.py:14:1: F401 'time' imported but unused
./src/bindings/python/flux/accounting/flux-account.py:18:1: F401 'pandas as pd' imported but unused
./src/bindings/python/flux/accounting/flux-account.py:27:1: C901 'main' is too complex (16)
./src/bindings/python/flux/accounting/print_hierarchy.py:12:1: F401 'sqlite3' imported but unused
./src/bindings/python/flux/accounting/print_hierarchy.py:19:1: E302 expected 2 blank lines, found 1
./src/bindings/python/flux/accounting/accounting_cli_functions.py:14:1: F401 'pwd' imported but unused
./src/bindings/python/flux/accounting/accounting_cli_functions.py:15:1: F401 'csv' imported but unused
./src/bindings/python/flux/accounting/accounting_cli_functions.py:69:5: F841 local variable 'last_parent_bank_seen' is assigned to but never used
./src/bindings/python/flux/accounting/job_archive_interface.py:12:1: F401 'sqlite3' imported but unused
./src/bindings/python/flux/accounting/job_archive_interface.py:84:5: F841 local variable 'hostname_arr' is assigned to but never used
./src/bindings/python/flux/accounting/job_archive_interface.py:178:58: E711 comparison to None should be 'if cond is not None:'
./src/bindings/python/flux/accounting/create_db.py:13:1: F401 'pandas as pd' imported but unused
./src/bindings/python/flux/accounting/create_db.py:15:1: F401 'argparse' imported but unused
./src/bindings/python/flux/accounting/create_db.py:19:1: E302 expected 2 blank lines, found 1
./src/bindings/python/flux/accounting/test/test_job_archive_interface.py:15:1: F401 'pandas as pd' imported but unused
./src/bindings/python/flux/accounting/test/test_job_archive_interface.py:18:1: F401 'flux.accounting.create_db as c' imported but unused
./src/bindings/python/flux/accounting/test/test_create_db.py:41:13: F841 local variable 'table' is assigned to but never used
./src/bindings/python/flux/accounting/test/test_create_db.py:73:9: F841 local variable 'cursor' is assigned to but never used
./src/bindings/python/flux/accounting/test/test_create_db.py:88:9: F841 local variable 'cursor' is assigned to but never used
./src/bindings/python/flux/accounting/test/test_example.py:13:1: F401 'sys' imported but unused
./src/bindings/python/flux/accounting/test/test_example.py:14:1: F401 'os' imported but unused
./src/bindings/python/flux/accounting/test/test_example.py:25:1: E305 expected 2 blank lines after class or function definition, found 1
1     C901 'main' is too complex (16)
1     E261 at least two spaces before inline comment
4     E302 expected 2 blank lines, found 1
2     E305 expected 2 blank lines after class or function definition, found 1
2     E401 multiple imports on one line
5     E402 module level import not at top of file
1     E711 comparison to None should be 'if cond is not None:'
12    F401 'pwd' imported but unused
5     F841 local variable 'last_parent_bank_seen' is assigned to but never used
33
reformatted /home/runner/work/flux-accounting/flux-accounting/config/tap-driver.py
reformatted /home/runner/work/flux-accounting/flux-accounting/src/bindings/python/flux/accounting/create_db.py
reformatted /home/runner/work/flux-accounting/flux-accounting/src/bindings/python/flux/accounting/accounting_cli_functions.py
reformatted /home/runner/work/flux-accounting/flux-accounting/src/bindings/python/flux/accounting/flux-account.py
reformatted /home/runner/work/flux-accounting/flux-accounting/src/bindings/python/flux/accounting/test/test_create_db.py
reformatted /home/runner/work/flux-accounting/flux-accounting/src/bindings/python/flux/accounting/test/test_example.py
reformatted /home/runner/work/flux-accounting/flux-accounting/src/bindings/python/flux/accounting/test/test_bank_subcommands.py
reformatted /home/runner/work/flux-accounting/flux-accounting/src/bindings/python/flux/accounting/test/test_user_subcommands.py
reformatted /home/runner/work/flux-accounting/flux-accounting/src/bindings/python/flux/accounting/test/test_job_archive_interface.py
All done! ✨ 🍰 ✨
9 files reformatted, 7 files left unchanged.
```
Perhaps the checks need to be updated to use autotools?

I'm still posting this PR as a WIP in case someone else can take a look.

### CODE REVIEW

**Comment by SteVwonder** [comment]
> It also looks like pylint is catching errors and black is reformatting code, but that isn't caught as an error in the actions(?)

Oof! That's not good.  Thanks for catching that!

> Perhaps the checks need to be updated to use autotools?

Yeah.  We decided to kick that can down the road a bit.  I realize we don't have an open issue for it though, so I opened on to better communicate that: https://github.com/flux-framework/flux-accounting/issues/82

**Comment by SteVwonder** [comment]
Thanks for all the work on this @grondo!

> I'm still posting this PR as a WIP in case someone else can take a look.

Do you mind if I (or someone else) pushes commits to this PR via your branch after they take a look?

**Comment by grondo** [comment]
Yeah, no problem, that is why I pushed it here!

**Comment by cmoussa1** [comment]
OK, just pushed a change to the `Lint with flake8` step that checks for the following:

The flake8 step currently reports warnings and errors but does not catch it as an error in GitHub actions, so remove the `--exit-zero` argument from the flake8 line and add a number of ignored warnings and error codes.

The `black` formatter step reports formatting changes in the Python files but also does not catch it as an error in GitHub actions, so add a specific version of the `black` formatter and add a `--check` argument to return a non-zero exit code if there would be formatting changes to files.

---

Now that I've edited these checks, Github actions should now be correctly reporting formatting errors about some of the files in this repo:

```
./src/bindings/python/flux/accounting/print_hierarchy.py:12:1: F401 'sqlite3' imported but unused
./src/bindings/python/flux/accounting/flux-account.py:14:1: F401 'time' imported but unused
./src/bindings/python/flux/accounting/flux-account.py:18:1: F401 'pandas as pd' imported but unused
./src/bindings/python/flux/accounting/accounting_cli_functions.py:14:1: F401 'pwd' imported but unused
./src/bindings/python/flux/accounting/accounting_cli_functions.py:15:1: F401 'csv' imported but unused
./src/bindings/python/flux/accounting/accounting_cli_functions.py:69:5: F841 local variable 'last_parent_bank_seen' is assigned to but never used
./src/bindings/python/flux/accounting/create_db.py:13:1: F401 'pandas as pd' imported but unused
./src/bindings/python/flux/accounting/create_db.py:15:1: F401 'argparse' imported but unused
./src/bindings/python/flux/accounting/job_archive_interface.py:12:1: F401 'sqlite3' imported but unused
./src/bindings/python/flux/accounting/job_archive_interface.py:84:5: F841 local variable 'hostname_arr' is assigned to but never used
./src/bindings/python/flux/accounting/test/test_job_archive_interface.py:15:1: F401 'pandas as pd' imported but unused
./src/bindings/python/flux/accounting/test/test_job_archive_interface.py:18:1: F401 'flux.accounting.create_db as c' imported but unused
./src/bindings/python/flux/accounting/test/test_create_db.py:41:13: F841 local variable 'table' is assigned to but never used
./src/bindings/python/flux/accounting/test/test_create_db.py:73:9: F841 local variable 'cursor' is assigned to but never used
./src/bindings/python/flux/accounting/test/test_create_db.py:88:9: F841 local variable 'cursor' is assigned to but never used
./src/bindings/python/flux/accounting/test/test_example.py:13:1: F401 'sys' imported but unused
./src/bindings/python/flux/accounting/test/test_example.py:14:1: F401 'os' imported but unused
./src/bindings/python/flux/accounting/test/test_example.py:25:1: E305 expected 2 blank lines after class or function definition, found 1
./config/tap-driver.py:9:29: E261 at least two spaces before inline comment
./config/tap-driver.py:13:50: E226 missing whitespace around arithmetic operator
./config/tap-driver.py:25:1: E305 expected 2 blank lines after class or function definition, found 1
1     E226 missing whitespace around arithmetic operator
1     E261 at least two spaces before inline comment
2     E305 expected 2 blank lines after class or function definition, found 1
12    F401 'pwd' imported but unused
5     F841 local variable 'last_parent_bank_seen' is assigned to but never used
```

Would it be okay to push the necessary fixes for these flagged Python files in additional commits to this PR?

**Comment by grondo** [comment]
> Would it be okay to push the necessary fixes for these flagged Python files in additional commits to this PR?

Yes, that is very reasonable. I would put the commits that fix all the failing checks first, then enable CI. (just more logical that way)

File: .github/workflows/main.yml
Comment (grondo) [inline .github/workflows/main.yml:36]: If you split black and flake8 into separate "steps" you would get a couple benefits:

 - the steps can run in parallel (probably not a big deal since they're both fast)
 - the GH workflow interface will show you which of flake8 or black formatting failed

Comment (cmoussa1) [reply .github/workflows/main.yml:36]: Thanks for pointing this out. I will fix the commit I pushed and separate the two into different steps.

**Comment by cmoussa1** [comment]
OK, just pushed a bunch of tiny formatting (both `flake8` and `black`) commits before my GitHub Actions commit that should fix all the errors that both `flake8` and `black` were reporting. For now, I left them as separate commits, but if you feel it would be best to squash them down into just a couple of commits, that would be fine with me as well.

File: .github/workflows/main.yml
Comment (SteVwonder) [inline .github/workflows/main.yml:22]: Do we still need to test against all three versions of python?  I'm assuming one would be sufficient now that this repo is no longer purely python. Thoughts?

**Review by SteVwonder — APPROVED** [review]
LGTM! Just one question below about the python version matrix.

File: .github/workflows/main.yml
Comment (cmoussa1) [reply .github/workflows/main.yml]: Thanks @SteVwonder; I agree. I just added a commit that drops the checks for `3.6` and `3.8`.

**Comment by cmoussa1** [comment]
Am I OK to set MWP on this PR?

**Comment by cmoussa1** [comment]
Going to set MWP on this PR in a little bit unless there are objections.

**Comment by grondo** [comment]
This might have to be manually merged since it touches the Github workflow yaml files...

**Comment by grondo** [comment]
A couple things to check before you merge this:

 * I'd consider giving the workflow a name so that it is more easily referenced. Otherwise, the "name" is the full path to the workflow yaml within the repo. You can set a name with the `name:` key at the top level
 * Make sure the mergify.yml is update with the correct names for any required `status-success` checks.

**Comment by cmoussa1** [comment]
Thanks @grondo - I just pushed two additional commits: one that adds a name to the entire workflow (I just copied the name from flux-core, `ci`), and I updated the `status-success` checks in **mergify.yml** (just removed the Python version checks that are not needed and replaced them with the flake8 and Black formatting checks).

File: .mergify.yml
Comment (grondo) [inline .mergify.yml:7]: actually I *think* the `black` and `flake8` checks will both be reported under the same status since they are serial checks run under the `run-checks:` job. If you want to split these out into separate checks then you can do so by moving each to a new "job" (at the same level as `run-checks` and `check-pr`.)

If you are fine with checks running serially, then you can just update the status name to
```suggestion
      - status-success="run-checks (3.7)"
```

Comment (cmoussa1) [reply .mergify.yml:7]: Ah, okay. I think I'm fine with the checks just running serially. I'll go ahead and push a change to update the status name to `run checks (3.7)`.

**Comment by grondo** [comment]
LGTM! I can't approve my own PR (even though this branch has totally changed since I first pushed it), but I removed `WIP` from the title and you have @SteVwonder's approval, so feel free to click the merge button!

**Pull request merged**
